### PR TITLE
test ivn with cloudimgs

### PR DESCRIPTION
**Pull request by sherysheng**

### CODE REVIEW

**Comment by chenge3** [comment]
@xiar @weia5-emc could you help review this patch?

File: test/functional/test_ivn.py [hunk 1]
```diff
@@ -93,21 +104,42 @@ def _start_node(self, node_info):
...
-        return output
+        ssh = sshclient.SSH(host=ns_ip, username="ubuntu", password="password")
+        ssh.wait_for_host_up(300)
+        status, output = ssh.exec_command("ifconfig")
+        return ssh
+
+    def ping_peer(self, ssh, peer_ip):
+        status, output = ssh.exec_command("ping -I eth0 %s  -c 5" % peer_ip)
+        print output
+        self.assertIn("0% packet loss", output, "node connection passed!")
```
Comment (weia5-emc) [inline test/functional/test_ivn.py]: missing "not" in "node connection passed!" ❓

File: infrasim/cloud_img.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import os
+import string
+import random
+from infrasim import run_command
+
+
+# def load_backing_file we can add load backing file from cloud image offical website later
+
+def gen_qemuimg(boot_img):
+    status, output = run_command("qemu-img create -f qcow2 -o backing_file=/home/infrasim/jenkins/data/"
```
Comment (weia5-emc) [inline infrasim/cloud_img.py]: fixed path is not safe for user. ☠️ This is a package instead of a test case.

File: infrasim/cloud_img.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                                 "cloudimgs/user-data cloudimgs/meta-data cloudimgs/network-config".format(myseed_name))
+    return str(os.getcwd() + "/cloudimgs/") + myseed_name
+
+
+def clear_files():
+    status, output = run_command("sudo rm -rf cloudimgs")
+    print "status" + str(status)
+
+
+def id_generator(size):
```
Comment (weia5-emc) [inline infrasim/cloud_img.py]: it's better to change variable name from "size" to "id_length".

File: infrasim/cloud_img.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+import random
+from infrasim import run_command
+
+
+# def load_backing_file we can add load backing file from cloud image offical website later
+
+def gen_qemuimg(boot_img):
+    status, output = run_command("qemu-img create -f qcow2 -o backing_file=/home/infrasim/jenkins/data/"
+                                 "ubuntu-16.04-server-cloudimg-amd64-120G.org.bak cloudimgs/{}".format(boot_img))
+    return str(os.getcwd() + "/cloudimgs/") + boot_img
```
Comment (weia5-emc) [inline infrasim/cloud_img.py]: force to use current folder is not a good choice. it's better to let user to specify it.

File: infrasim/cloud_img.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    print "status" + str(status)
+
+
+def id_generator(size):
+    return ''.join(random.choice(string.ascii_lowercase + string.digits) for _ in range(size))
+
+
+def create_network_config_file(mac_addr, guest_ip, gate_way, mac1):
+    script_name = str(os.getcwd()) + "/cloudimgs/network-config"
+    script_content = '''---
```
Comment (weia5-emc) [inline infrasim/cloud_img.py]: strongly suggest to use dictionary here instead of multi-line string.

File: test/functional/test_ivn.py [hunk 6]
```diff
@@ -10,16 +10,28 @@
...
 from test.fixtures import FakeConfig
 from infrasim.model import CNode
 from infrasim import sshclient
+from infrasim import cloud_img
 from test import fixtures
 
 old_path = os.environ.get('PATH')
 new_path = '{}/bin:{}'.format(os.environ.get('PYTHONPATH'), old_path)
+status, output = run_command("mkdir cloudimgs")
+global a_boot_image
```
Comment (weia5-emc) [inline test/functional/test_ivn.py]: 4 lines of global declaration are useless.

File: test/functional/test_ivn.py [hunk 7]
```diff
@@ -93,21 +104,42 @@ def _start_node(self, node_info):
...
+            fake_node["compute"]["networks"][0]["mac"] = "00:60:16:93:b9:2a"
+        if "test1" in node_name:
+            fake_node['compute']['cdrom'] = {}
+            fake_node['compute']['cdrom']['file'] = iso
+            fake_node["compute"]["networks"].append({"device": "e1000", "network_mode": "bridge", "network_name": "br0",
+                                                     "mac": "00:60:16:93:b9:1a"})
+            fake_node["compute"]["networks"][0]["port_forward"] = [{"outside": 8022, "inside": 22, "protocal": "tcp"}]
+            fake_node["compute"]["networks"][0]["mac"] = "00:60:16:93:b9:2d"
+        print "fake_node"
+        print fake_node
```
Comment (weia5-emc) [inline test/functional/test_ivn.py]: it's better to remove "print".  print should be debug purpose only.

File: test/functional/test_ivn.py [hunk 8]
```diff
@@ -93,21 +104,42 @@ def _start_node(self, node_info):
...
         return fake_node_up
 
     def client_ssh(self, ns_ip):
-        ssh = sshclient.SSH(host=ns_ip, username="root", password="root", port=8022)
-        ssh.wait_for_host_up()
-        status, output = ssh.exec_command("ls")
-        return output
+        ssh = sshclient.SSH(host=ns_ip, username="ubuntu", password="password")
+        ssh.wait_for_host_up(300)
+        status, output = ssh.exec_command("ifconfig")
```
Comment (weia5-emc) [inline test/functional/test_ivn.py]: it seems this line is useless.

File: test/functional/test_ivn.py (hunk 7 above)
Comment (sherysheng) [reply test/functional/test_ivn.py]: okay!

File: infrasim/cloud_img.py (hunk 2 above)
Comment (sherysheng) [reply infrasim/cloud_img.py]: I use this temp path for test purpose only, the folder will be rm after test

File: test/functional/test_ivn.py (hunk 1 above)
Comment (sherysheng) [reply test/functional/test_ivn.py]: There should be "0% packet loss" in the test result.

File: infrasim/cloud_img.py (hunk 3 above)
Comment (sherysheng) [reply infrasim/cloud_img.py]: good point! I will modify the variable name

File: infrasim/cloud_img.py (hunk 4 above)
Comment (sherysheng) [reply infrasim/cloud_img.py]: It's a tmp folder, user should not care the path.

**Pull request merged**